HackIllinois/ViewControllers/HIProfileViewController.swift

```diff
@@ -248,11 +254,21 @@ extension HIProfileViewController {
 
     func updateProfile() {
         guard let profile = HIApplicationStateController.shared.profile else { return }
+        guard let user = HIApplicationStateController.shared.user else { return }
         view.layoutIfNeeded()
 
         if let url = URL(string: profile.avatarUrl), let imgValue = HIConstants.PROFILE_IMAGES[url.absoluteString] {
             profilePictureView.changeImage(newImage: imgValue)
         }
+        // Generate a QR Code with the user's id
+        if let logo = UIImage(named: "QR Code Logo"), let whiteBackground = UIImage(named: "QR Code Background") {
+            // Custom QR Code with logo
+            self.qrImageView.image = UIImage.init(qrString: user.id, qrCodeColor: (\HIAppearance.qrTint).value, backgroundImage: whiteBackground, logo: logo)
+        } else {
+            // Regular, black QR Code
+            let size = max(UIScreen.main.bounds.width, UIScreen.main.bounds.height)
+            self.qrImageView.image = UIImage.init(qrString: user.id, size: size)
+        }
         profileNameView.text = profile.firstName + " " + profile.lastName
```

Comment:
Loading time of profile takes too long due to QR code generation. Perhaps the code generation (lines 263-271) can be done after the rest of the profile has loaded (viewDidAppear) or asynchronously?